tests/csapi/room_upgrade_test.go

```diff
@@ -0,0 +1,309 @@
+package csapi_tests
+
+import (
+	"testing"
+	"time"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+// sytest: /upgrade creates a new room
+func TestUpgradeRoom(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+	roomID := alice.CreateRoom(t, map[string]interface{}{
+		"preset": "public_chat",
+	})
+
+	_, tombstoneSinceToken := alice.MustSync(t, client.SyncReq{TimeoutMillis: "0"})
+	_, sinceToken := alice.MustSync(t, client.SyncReq{TimeoutMillis: "0"})
```

Comment:
A bit odd to sync twice rather than copy the token into a new var

Reply:
Might be clearer if this was a `MustSyncUntil` which waits for Alice to see her membership of the new room?

Reply:
Using the same since-token creates - for my feeling - Undefined Behaviour, and these exist to detect two different things in disjoint "timelines";
- One is looking for the tombstone event, which can happen before or after creating the new room.
- One is looking for the new room, which can happen before or after the tombstone event is emitted.

This might look like a clear case of using the same since-token twice, but I've encountered subtle weird behaviour in the past with using the same sync token twice, and as this kind of behaviour is not explicitly defined in the spec (it only talks about and assumes that one uses a since token exactly once), i'm being conservative here, creating two different since tokens to use for different "spans of time" to test.

Reply:
Okay. Can you please spell this out with a comment: something to the effect of "the tombstone and new room can appear down /sync in either order, so we check for these using separate /sync tracks". Can we also have a more descriptive name for the second `sinceToken` which describes what it is being used to wait for, please.